recipe/build-mpi.sh

```diff
@@ -105,13 +104,19 @@ if [[ "$target_platform" == linux-* ]]; then
     export BASH_SHELL="/bin/bash"
 fi
 
+# allow argument mismatch in Fortran
+# https://github.com/pmodels/mpich/issues/4300
+export FFLAGS="$FFLAGS -fallow-argument-mismatch"
+export FCFLAGS="$FCFLAGS -fallow-argument-mismatch"
```

Comment:
```
x86_64-apple-darwin13.4.0-gfortran: error: unrecognized command line option '-fallow-argument-mismatch'; did you mean '-Wno-argument-mismatch'?
```

Reply:
So looks like we need these flags only for macOS on newer architectures. Wonderful. Could you please remember me why I do this to myself?

Reply:
```
gfortran_linux-64:       10.3.0-hb09a455_6
gfortran_osx-64:          9.3.0-h18f7dce_15
gfortran_osx-arm64:      11.0.1.dev0-h57527a5_15
```

Reply:
Looks like it worked?